src/Mvc/Mvc.Core/src/ModelBinding/FormValueProviderFactory.cs

```diff
@@ -33,9 +36,20 @@ public Task CreateValueProviderAsync(ValueProviderFactoryContext context)
         private static async Task AddValueProviderAsync(ValueProviderFactoryContext context)
         {
             var request = context.ActionContext.HttpContext.Request;
+            IFormCollection form;
+            try
+            {
+                form = await request.ReadFormAsync();
+            }
+            catch (InvalidDataException ex)
+            {
+                form = FormCollection.Empty;
+                context.ActionContext.ModelState.TryAddModelException(string.Empty, ex);
```

Comment:
Using `string.Empty` here is really problematic, but also incorrect in this case. Can we explore the other alternative?

Reply:
Maybe it shouldn't be ModelState at all but maybe something different to add error. Eventually it's not an issue with modelstate but rather the whole request.